Sources/TuistKit/Generator/WorkspaceGenerator.swift

```diff
@@ -59,22 +64,59 @@ final class WorkspaceGenerator: WorkspaceGenerating {
         let workspacePath = workspaceRootPath.appending(component: workspaceName)
         let workspaceData = XCWorkspaceData(children: [])
         let workspace = XCWorkspace(data: workspaceData)
+        
+        /*-----------------------------------------------------------------------------------------
+         // MARK: - Manifest Group
+         -----------------------------------------------------------------------------------------*/
 
+        let manifestFiles: [XCWorkspaceDataElement] = [ Manifest.workspace, Manifest.setup ]
+            .lazy
+            .map(\.fileName |> path.appending(component:))
+            .filter(fileHandler.exists)
+            .map(\.workspaceFileElement)
+
+        workspace.data.children.append(contentsOf: manifestFiles)
+        
+        /*-----------------------------------------------------------------------------------------
+         // MARK: - Projects
+         -----------------------------------------------------------------------------------------*/
+        
         try graph.projects.forEach { project in
             let sourceRootPath = try projectDirectoryHelper.setupProjectDirectory(project: project,
                                                                                   directory: directory)
             let generatedProject = try projectGenerator.generate(project: project,
                                                                  options: options,
                                                                  graph: graph,
                                                                  sourceRootPath: sourceRootPath)
-
+            
             let relativePath = generatedProject.path.relative(to: path)
-            let location = XCWorkspaceDataElementLocationType.group(relativePath.asString)
-            let fileRef = XCWorkspaceDataFileRef(location: location)
-            workspace.data.children.append(XCWorkspaceDataElement.file(fileRef))
+            workspace.data.children.append(relativePath.workspaceFileElement)
         }
+        
         try workspace.write(path: workspacePath.path, override: true)
 
         return workspacePath
     }
+    
+}
+
+/*-----------------------------------------------------------------------------------------
+ // MARK: - Helper for generating XCWorkspaceDataElement from AbsolutePath and RelativePath
+ -----------------------------------------------------------------------------------------*/
+
+private protocol WorkspaceFileElement {
+    var asString: String { get }
 }
+
+extension WorkspaceFileElement {
+    
+    var workspaceFileElement: XCWorkspaceDataElement {
+        let location = XCWorkspaceDataElementLocationType.group(asString)
+        let fileRef = XCWorkspaceDataFileRef(location: location)
+        return .file(fileRef)
+    }
+    
+}
+
+extension AbsolutePath: WorkspaceFileElement { }
```

Comment:
I find it confusing that a the path types that are generic get the `workspaceFileElement` method that is specific in the context of a workspace. I'd find the way of narrowing the scope of the approach or rather finding an alternative.

Reply:
Yes, I agree - let's improve this.